pillarbox-core-business/src/main/java/ch/srgssr/pillarbox/core/business/exception/BlockReasonException.kt

```diff
@@ -12,87 +12,109 @@ import kotlinx.datetime.Instant
 import java.io.IOException
 
 /**
- * Block reason exception
+ * An exception thrown when a [Chapter] is blocked for a specific reason. The specific reason is indicated by the type of [BlockReasonException]
+ * thrown.
+ *
+ * Each subclass of [BlockReasonException] corresponds to a specific [BlockReason] and provides a [messageResId] containing a user-friendly localized
+ * message describing the block reason.
+ *
+ * **Checking the blocking reason**
+ *
+ * ```kotlin
+ * val exception: BlockReasonException
+ * when (exception) {
+ *     // Chapter is geo-blocked.
+ *     is BlockReasonException.GeoBlock -> TODO()
```

Comment:
Not sure that `TODO()` is the best placeholder for a missing implementation in a documentation, excep if this is common practice I am not aware of. Feels like the documentation is still WIP when you read it.

Reply:
I've replaced the `TODO()` with a log and removed the corresponding comments. Let me know if you had something else in mind.